### ISSUE TITLE
[cvss]: Unable to parse vectors with temporal score component

### ISSUE THREAD

**Issue by ctron** [rustsec/rustsec#1085]
Parsing scores from the `cvelistV5` repository make some scores fail with:

```
unknown CVSS metric name: `E`
```

It looks like there's a todo remaining for this: https://github.com/rustsec/rustsec/blob/f79b59b45dc7b04f584c524b4c482dffce7f158a/cvss/src/v3.rs#L5

I might be able to work on this and provide a PR if someone can review it.

**Comment by tarcieri** [comment]
Yep, we currently only support the CVSS v3.1 Base metric group, hence the type being `cvss::v3::Base`. The Temporal and Environmental metric groups are separate and currently unimplemented.

All that said, it would probably be good to add CVSS v4.0 support first.

**Comment by bxf12315** [comment]
@tarcieri @ctron , [PR](https://github.com/rustsec/rustsec/pull/1198) try to fix the issue, but I got many compile errors from other components. Could you please help me fix it?

**Comment by bxf12315** [comment]
@tarcieri @ctron  according to the PR's comment, I refactored the PR, let it will not affect other modules. Please check again.

**Comment by bxf12315** [comment]
@tarcieri Do you have any plans to upgrade to 4.0? I think if the relevant models could be separated into their own repository, it would be easier for collaborative development.

**Comment by tarcieri** [comment]
#1087 is the tracking issue for CVSS v4.0

I don't see any reason why it can't be added to the `cvss` crate. The existing module structure is already designed to accommodate new versions.
